src/include/miopen/interpolate/problem_description.hpp

```diff
@@ -0,0 +1,285 @@
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace interpolate {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& scaleFactorsDesc_,
+                       const miopenInterpolateMode_t mode_,
+                       const bool align_corners_,
+                       bool is_fwd_)
+        : scaleFactorsDesc(scaleFactorsDesc_),
+          mode(mode_),
+          align_corners(align_corners_),
+          is_fwd(is_fwd_)
+    {
+        IsValidMode();
+    }
+
+    const TensorDescriptor& GetScaleFactorsDesc() const { return scaleFactorsDesc; }
+    miopenInterpolateMode_t GetMode() const { return mode; }
+    bool GetAlignCorners() const { return align_corners; }
+
+    bool IsValidMode() const
+    {
+        if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST && mode != MIOPEN_INTERPOLATE_MODE_LINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BILINEAR && mode != MIOPEN_INTERPOLATE_MODE_TRILINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BICUBIC)
+        {
+            std::cout << "MODE: " << mode << std::endl;
+            MIOPEN_THROW(miopenStatusBadParm, "Interpolate: Invalid mode.");
+        }
+        return true;
+    }
+
+    bool IsValidStride(TensorDescriptor td) const
+    {
+        auto strides = td.GetStrides();
+        auto lengths = td.GetLengths();
+        std::vector<std::pair<size_t, size_t>> p;
+        p.reserve(td.GetSize());
+        std::transform(strides.begin(),
+                       strides.end(),
+                       lengths.begin(),
+                       std::back_inserter(p),
+                       [](size_t a, size_t b) { return std::make_pair(a, b); });
+        std::sort(p.begin(), p.end());
+        for(int i = 1; i < p.size(); ++i)
+        {
+            if(p[i].first != p[i - 1].first * p[i - 1].second)
+                MIOPEN_THROW(miopenStatusBadParm, "Interpolate: Tensor strides do not valid.");
+        }
+        return true;
+    }
+
+protected:
+    TensorDescriptor scaleFactorsDesc;
+    miopenInterpolateMode_t mode;
+    bool align_corners = false;
+    bool is_fwd;
+
+    NetworkConfig MakeForwardNetworkConfig() const;
+};
+
+struct FwdProblemDescription : ProblemDescription
+{
+    FwdProblemDescription(const TensorDescriptor& inputDesc_,
+                          const TensorDescriptor& outputDesc_,
+                          const TensorDescriptor& scaleFactorsDesc_,
+                          const miopenInterpolateMode_t mode_,
+                          const bool align_corners_)
+        : ProblemDescription(scaleFactorsDesc_, mode_, align_corners_, true)
+    {
+        inputDesc  = inputDesc_;
+        outputDesc = outputDesc_;
+        IsValidDims();
+        IsValidLength();
+        IsAllValidStride();
+    }
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+
+    bool IsValidLength() const
```

Comment:
I made these two separate functions because IsValidDims is for validating dimensions according to the mode, and IsValidLength is for general valid input of the op.